### PR TITLE
8588 - BUG - Migration to remove noticeOfTrialDate when no NTD

### PR DESCRIPTION
**Pull request by matthopson**

### CODE REVIEW

**Review by kkoskelin — APPROVED** [review]

File: web-api/migration-terraform/main/lambdas/migrations/bug-0039-notice-of-trial-date.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+} = require('../../../../../shared/src/business/entities/EntityConstants');
+const applicationContext = createApplicationContext({});
+
+const migrateItems = async (items, documentClient) => {
+  const itemsAfter = [];
+
+  for (const item of items) {
+    if (item.pk.startsWith('case|') && item.sk.startsWith('case|')) {
+      const fullCase = await documentClient
+        .query({
```
Comment (mmarcotte) [inline web-api/migration-terraform/main/lambdas/migrations/bug-0039-notice-of-trial-date.js]: I think there are some cases that are too big for this query alone. You might not get back all of the information you desire. You may have to paginate this query with `.queryFull`

Comment (matthopson) [reply web-api/migration-terraform/main/lambdas/migrations/bug-0039-notice-of-trial-date.js]: @mmarcotte I wonder if there have been issues with past migrations that use this pattern. We've done this a lot in the past (used `query` to fetch all case items).

Comment (mmarcotte) [reply web-api/migration-terraform/main/lambdas/migrations/bug-0039-notice-of-trial-date.js:19]: mayyyybe

**Review by codyseibert — APPROVED** [review]

File: web-api/migration-terraform/main/lambdas/migration-segments.js [hunk 2]
```diff
@@ -69,6 +72,11 @@ const migrateRecords = async ({
     items = await migration0038(items);
   }
 
+  if (!ranMigrations['bug-0039-notice-of-trial-date.js']) {
+    applicationContext.logger.debug('about to run devex migration 0039');
```
Comment (sutt0n) [inline web-api/migration-terraform/main/lambdas/migration-segments.js]: devex migration or bug migration?

File: web-api/migration-terraform/main/lambdas/utilities.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const queryFullCase = async (documentClient, docketNumber) => {
```
Comment (sutt0n) [inline web-api/migration-terraform/main/lambdas/utilities.js:1]: 🎉

**Review by sutt0n — APPROVED** [review]

File: web-api/migration-terraform/main/lambdas/migration-segments.js (hunk 2 above)
Comment (matthopson) [reply web-api/migration-terraform/main/lambdas/migration-segments.js]: Ayyyy, good catch mang

**Pull request merged**